### PR TITLE
Base integrations testing class

### PR DESCRIPTION
**Pull request by KSGulin**
Base integrations testing class on which each integrations test suite will be based.

Integrations:

- [ ] Yolov5 - #747 
- [ ] Image Classification #794
- [ ] Transformers [Not yet started]

### CODE REVIEW

**Comment by dbogunowicz** [comment]
@KSGulin is it ready for the review?

**Comment by KSGulin** [comment]
@dbogunowicz It is, but probably best to wait until after the team briefing to make sure any feedback can be incorporated upfront

File: tests/integrations/base_tester.py [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+        self.cadence = raw_config.pop("cadence")
+
+        # Command types present in this config
+        self.command_types = [_type for _type in raw_config]
+
+        # Final command stub
+        self.command_stubs_final = self.get_root_commands(raw_config)
+
+        # Map each present command type to the pydantic class and pre-args (see below)
+        self.configs = {
```
Comment (bfineran) [inline tests/integrations/base_tester.py:102]: thoughts on creating a separate class to handle the loading/parsing of config files? can also include documentation on how to write configs there. happy to take this on as well in an update 🙂

**Review by bfineran — APPROVED** [review]
LGTM overall, one thing I'd suggest us adding in is some common test functions to the base class that integrations should implement at minimum.

File: tests/integrations/base_tester.py [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+            os.environ.get("NM_TEST_CADENCE", "commit"), os.path.dirname(__file__)
+        ),
+        scope="class",
+    )
+    def integration_manager(request):
+        """
+        Fixture with a lifecycle of:
+        - Create integration manager instance (child of BaseIntegrationManager)
+        - Yield manager to test
+        - teardown
```
Comment (dbogunowicz) [inline tests/integrations/base_tester.py:237]:
```suggestion
        - Teardown
```

**Review by dbogunowicz — APPROVED** [review]
Please fix the commit history, otherwise LGTM.
